configs/projects/bolt-runtime.rb

```diff
@@ -0,0 +1,138 @@
+project 'bolt-runtime' do |proj|
+  # Used in component configurations to conditionally include dependencies
+  proj.setting(:runtime_project, 'bolt')
+  proj.setting :ruby_version, '2.4.3'
+  platform = proj.get_platform
+
+  proj.version_from_git
+  proj.generate_archives true
+  proj.generate_packages false
+
+  proj.description "The Bolt runtime contains third-party components needed for Bolt standalone packaging"
+  proj.license "See components"
+  proj.vendor "Puppet, Inc.  <info@puppet.com>"
+  proj.homepage "https://puppet.com"
+  proj.identifier "com.puppetlabs"
+
+  if platform.is_windows?
+    proj.setting(:company_id, "PuppetLabs")
+    proj.setting(:product_id, "Bolt")
+    if platform.architecture == "x64"
+      proj.setting(:base_dir, "ProgramFiles64Folder")
+    else
+      proj.setting(:base_dir, "ProgramFilesFolder")
+    end
+    # We build for windows not in the final destination, but in the paths that correspond
+    # to the directory ids expected by WIX. This will allow for a portable installation (ideally).
+    proj.setting(:prefix, File.join("C:", proj.base_dir, proj.company_id, proj.product_id))
+  else
+    proj.setting(:prefix, "/opt/puppetlabs/bolt")
+  end
+
+  proj.setting(:ruby_dir, proj.prefix)
+  proj.setting(:bindir, File.join(proj.prefix, 'bin'))
+  proj.setting(:ruby_bindir, proj.bindir)
+  proj.setting(:libdir, File.join(proj.prefix, 'lib'))
+  proj.setting(:includedir, File.join(proj.prefix, "include"))
+  proj.setting(:datadir, File.join(proj.prefix, "share"))
+  proj.setting(:mandir, File.join(proj.datadir, "man"))
+
+  if platform.is_windows?
+    proj.setting(:host_ruby, File.join(proj.ruby_bindir, "ruby.exe"))
+    proj.setting(:host_gem, File.join(proj.ruby_bindir, "gem.bat"))
+
+    # For windows, we need to ensure we are building for mingw not cygwin
+    platform_triple = platform.platform_triple
+    host = "--host #{platform_triple}"
+  else
+    proj.setting(:host_ruby, File.join(proj.ruby_bindir, "ruby"))
+    proj.setting(:host_gem, File.join(proj.ruby_bindir, "gem"))
+  end
+
+  ruby_base_version = proj.ruby_version.gsub(/(\d)\.(\d)\.(\d)/, '\1.\2.0')
+  proj.setting(:gem_home, File.join(proj.libdir, 'ruby', 'gems', ruby_base_version))
+  proj.setting(:gem_install, "#{proj.host_gem} install --no-rdoc --no-ri --local --bindir=#{proj.ruby_bindir}")
+
+  proj.setting(:platform_triple, platform_triple)
+  proj.setting(:host, host)
+
+  proj.setting(:artifactory_url, "https://artifactory.delivery.puppetlabs.net/artifactory")
+  proj.setting(:buildsources_url, "#{proj.artifactory_url}/generic/buildsources")
+
+  # Define default CFLAGS and LDFLAGS for most platforms, and then
+  # tweak or adjust them as needed.
+  proj.setting(:cppflags, "-I#{proj.includedir} -I/opt/pl-build-tools/include")
```

Comment:
There's some refactoring that could be done to bring pdk inline with this and puppet-agent, and extract a `base-ruby` file for handling some of these settings.